php/libraries/User.class.inc

```diff
@@ -541,5 +541,24 @@ class User extends UserPermissions
         return $this->hasPermission($code)
             && $this->hasCenter($center_id);
     }
+
+    /**
+     * Determines if the user has ever logged in successfully
+     *
+     * @return bool
+     */
+    public function hasEverLoggedIn(): bool
+    {
+        $count = (\Database::singleton())->pselectOne(
+            "SELECT
+               COUNT(1)
+             FROM user_login_history 
```

Comment:
I feel like because we are querying the login_history instead of the `user` table that we should atleast check if the user exists?

Reply:
I guess in the scope of the hasLoggedIn() function, querying the history table is fine. I was just skeptical removing the Loris Except in canRejectAccount, and then querying the history table instead of the users table instead but we established above that a $user instance would need to exist in the first place and so a user exists check is unneccesary